frontend/src/pages/Restaurants.js

```diff
@@ -1,25 +1,106 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { VerticalFilter } from '../components/filtration/VerticalFilter';
 import RestaurantItem from '../components/restaurants/RestaurantItem';
 import Select from 'react-select'
 import Navbar from '../components/navbar/Navbar';
 import SelectStyle from '../components/search/SelectStyle';
 import SelectLogic from '../components/search/SelectLogic';
 import './Restaurants.css'
-import { RestaurantPhotos } from
-  '../components/restaurants/PhotoSlider/RestaurantPhotos';
-import RestaurantPagination from '../components/restaurants/pagination/Pagination';
-
+import RestaurantPagination from
+  '../components/restaurants/pagination/Pagination';
+import { ImagePlaceHolder } from
+  '../components/restaurants/PhotoSlider/ImagePlaceHolder';
 
 export default function Restaurants() {
   const { customThemes, customStyles } = SelectStyle();
-  const { sortOptions, setSortResultHandler } = SelectLogic();
+  const { sortOptions, setSortResultHandler, } = SelectLogic();
+
+  const [priceRange, setPriceRange] = useState([]);
+  const [featured, setFeatured] = useState([]);
+
+  const [restaurants, setRestaurants] = useState([]);
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [restaurantsPerPage] = useState(5);
+
+  const indexOfLastRestaurant = currentPage * restaurantsPerPage;
+  const indexOfFirstRestaurant = indexOfLastRestaurant - restaurantsPerPage;
+
+  const paginate = (pageNumber) => {
+    setCurrentPage(pageNumber)
+  }
+
+  const handlePriceRangeFilters = (filters) => {
+    const newFilters = [...filters]
+    setPriceRange(newFilters)
+  }
+
+  const handleFeaturedFilters = (filters) => {
+    const newFilters = [...filters]
+    setFeatured(newFilters)
+  }
+
+  const priceRangeFilterValues = () => priceRange.map((item) => {
+        if (priceRange.indexOf(item) === priceRange.length - 1) {
+            return item + "&"
+        }
+        else {
+            return item
+        }
+    }
+  )
+
+  const featuredFilterValues = () => featured.map((item) => {
+        return item + "&"
+  })
+
+  var chosenPriceRanges= priceRangeFilterValues();
+  var chosenFeatured = featuredFilterValues();
+
+ const showFilteredResults = () => {
```

Comment:
The logic is mostly fine, some small problems:
* vegan and vegetarian only worked when some price was checked -> there is a duplicate if statement
* trailing & -> for example prague-college/restaurants?vegan&
* vegan & vegetarian joined with a comma instead of & -> prague-college/restaurants/?vegan,vegetarian

I found out about all of these from backend logs

This is what fixed the issues above for me in Restaurants.js:
```javascript
  const featuredFilterValues = () => featured.map((item) => {
        return item
  })

  var chosenPriceRanges= priceRangeFilterValues();
  var chosenFeatured = featuredFilterValues();

 const showFilteredResults = () => {
    var pragueCollegeRestaurants = "/prague-college/restaurants?"
    if (chosenPriceRanges.length === 0 & chosenFeatured.length === 0) {
      return pragueCollegeRestaurants
    }
    if (chosenPriceRanges.length !== 0 & chosenFeatured.length !== 0) {
      return pragueCollegeRestaurants +=
            "price-range=" + chosenPriceRanges.join(",") + chosenFeatured.join("&")
    }
    if (chosenPriceRanges.length !== 0) {
      return pragueCollegeRestaurants +=
            "price-range=" + chosenPriceRanges.join(",")
    }
    if (chosenFeatured.length !== 0) {
        return pragueCollegeRestaurants + chosenFeatured.join("&")
   }
  }
```

In the future this still would have to be redone, because this is mostly hardcoded around price-range and there will be many more filters to apply.

Also, the page is called restaurants but displays only restaurants around Prague College. Wouldn't it be better to rename it, so this is more obvious to the user?

Reply:
Alright, I will fix those problems. Yes, you're right, the filtration is currently hardcoded. Because of the deadline of basic functionality, I focused on implementing only these 2 filters. I will try to make it more dynamic after we pass the basic functionality milestone.

I think it will be better to rename the header of the restaurants page rather than changing the page name in navbar.
Like this : 
![image](https://user-images.githubusercontent.com/56120787/115850182-b82bce80-a436-11eb-8b0f-e97774dd7b22.png)

Reply:
> I will try to make it more dynamic after we pass the basic functionality milestone.

Sounds good.

> I think it will be better to rename the header of the restaurants page rather than changing the page name in navbar.

Agreed, I think this is good for now.